### PR TITLE
Added cpu model support #2

### LINKED ISSUE #98: CPU Model while using opennebula_virtual_machine
Hey guys I was wondering how can we specify CPU MODEL while using opennebula_virtual_machine . Can you please helo with the same ?

### PR DESCRIPTION
**Pull request by ankur-gupta-guavus**
@jaypif Please review this PR for adding cpu model support. This is from current master

### CODE REVIEW

**Comment by ankur-gupta-guavus** [comment]
@jaypif CI build is passing now. Please check and suggest next steps

File: opennebula/resource_opennebula_virtual_machine.go [hunk 1]
```diff
@@ -123,10 +123,25 @@ func resourceOpennebulaVirtualMachine() *schema.Resource {
...
 			},
-			"cpu":      cpuSchema(),
-			"vcpu":     vcpuSchema(),
-			"memory":   memorySchema(),
-			"context":  contextSchema(),
+			"cpu":     cpuSchema(),
+			"vcpu":    vcpuSchema(),
+			"memory":  memorySchema(),
+			"context": contextSchema(),
+			"cpumodel": {
```
Comment (jaypif) [inline opennebula/resource_opennebula_virtual_machine.go]: you use it in shared schema.
It must be declared in shared schema to be also used by VM templates

Can you move this part in sharedSchema ?

File: opennebula/shared_schemas.go [hunk 2]
```diff
@@ -353,6 +353,15 @@ func generateVMTemplate(d *schema.ResourceData, tpl *vm.Template) {
 		tpl.AddOS(vmk.Boot, osconfig["boot"].(string))
 	}
 
+	//Generate CPU Model definition
+	cpumodel, ok := d.Get("cpumodel").([]interface{})
```
Comment (jaypif) [inline opennebula/shared_schemas.go]: `cpumodel` does not exist for VM temapltes while this functon is also called in that case.

that is why everything linke to this feature must be wrote in "sharedSchema .go" directly

**Review by jaypif — CHANGES_REQUESTED** [review]

**Comment by ankur-gupta-guavus** [comment]
@jaypif There you go, i have added schema to shared_schemas.go. Also I have added a test case in template test files for cpumodel

File: opennebula/resource_opennebula_virtual_machine.go (hunk 1 above)
Comment (ankur-gupta-guavus) [reply opennebula/resource_opennebula_virtual_machine.go]: moved to shared_schemas.go

File: opennebula/shared_schemas.go (hunk 2 above)
Comment (ankur-gupta-guavus) [reply opennebula/shared_schemas.go]: Added in vm templates

**Review by jaypif — APPROVED** [review]

**Comment by jaypif** [comment]
Thanks this is perfect!

**Pull request merged**